client/layout/masterbar/logged-in.jsx

```diff
@@ -155,7 +163,7 @@ class MasterbarLoggedIn extends React.Component {
 				: getStatsPathForTab( 'day', siteSlug );
 
 		let mySitesUrl = domainOnlySite ? domainManagementList( siteSlug ) : homeUrl;
-		if ( config.isEnabled( 'nav-unification' ) && 'reader' !== section ) {
+		if ( config.isEnabled( 'nav-unification' ) && 'sites' === section ) {
```

Comment:
In case that we are already in my-sites section unset the URL so that clicking my-sites to open / close the sidebar doesn't navigate you away.